### ISSUE TITLE
Clarify whether or not a `Voice` should be paused following creation and ensure each backend abides by this.

### ISSUE THREAD

**Issue by mitchmindtree** [tomaka/cpal#175]
Currently the `beep` example calls `play` after constructing a voice, implying that upon creation it is paused by default which seems totally reasonable.

However, on Linux a `Voice` immediately begins playing during construction whether or not `play` is called.

Perhaps we should clarify that voices should be `paused` upon creation in the `build_voice` docs and fix this implementation in the ALSA backend.

**Comment by tomaka** [comment]
I had already thought about this problem, but I can't decide whether voices should be initially playing or initially paused.

**Comment by mitchmindtree** [comment]
I'm leaning towards paused by default, but don't really have a good reason as to why. I'm open to either, as long as we settle upon something for consistency. Maybe it's worth seeing what each of the backends do by default, and if they're all the same go with that? ALSA seems to play by default. I can't remember what coreaudio does off the top of my head.

**Comment by tomaka** [comment]
The argument I see in favour of pause by default is that you might want to create multiple voices at initialization without running the events loop immediately, in which case you'd prefer the voices to be paused.

However this is a minor use case, and other than that having to cast play every time you create a voice can be annoying, and if you forget to do it then you might not understand why things aren't working.

**Comment by tomaka** [comment]
Another thing to take into account is that if you create a play-by-default voice then pause it, you might have a problem because the backend attempted to get data between the creation of the voice and the pausing.

Alternatively we could add a `build_paused_voice` method.

**Comment by mitchmindtree** [comment]
> Alternatively we could add a build_paused_voice method.

Yeah I like this idea - I'd be happy for us to have `build_voice` return a playing voice and then have `build_paused_voice` call `build_voice` and `pause` immediately after?

**Comment by tomaka** [comment]
> Yeah I like this idea - I'd be happy for us to have build_voice return a playing voice and then have build_paused_voice call build_voice and pause immediately after?

Well, `build_paused_voice` would have to be implementation-specific. Otherwise you have the problem that I mentioned.

Alternatively, the opposite would work: `build_voice()` calls `build_paused_voice()` then `play()`.

**Comment by roderickvd** [comment]
`Voice` has become `Stream` and has play/pause semantics that everybody seems happy with.
